controller/appcontroller_test.go

```diff
@@ -1808,6 +1809,220 @@ func TestUnchangedManagedNamespaceMetadata(t *testing.T) {
+		if !newOK {
+			ctrl.appOperationQueue.AddRateLimited(key)
+		}
+		if ctrl.hydrator != nil {
+			ctrl.appHydrateQueue.AddRateLimited(newApp.QualifiedName())
+		}
+		ctrl.clusterSharding.UpdateApp(newApp)
+	}
+
+	// Helper to check if refresh was requested
+	// Note: isRefreshRequested uses appName directly, but requestAppRefresh uses toAppKey(appName)
+	// For qualified names (namespace/name), toAppKey returns the same value, so they should match
+	checkRefreshRequested := func(appName string, shouldBeRequested bool, msg string) {
+		// Use the same key format as requestAppRefresh
+		key := ctrl.toAppKey(appName)
+		ctrl.refreshRequestedAppsMutex.Lock()
+		_, isRequested := ctrl.refreshRequestedApps[key]
+		ctrl.refreshRequestedAppsMutex.Unlock()
+		assert.Equal(t, shouldBeRequested, isRequested, "%s: Refresh request state mismatch for app %s (key: %s)", msg, appName, key)
+	}
+
+	t.Run("Status-only update without annotation should NOT trigger refresh", func(_ *testing.T) {
+		// Reset refresh state
+		ctrl.refreshRequestedAppsMutex.Lock()
+		ctrl.refreshRequestedApps = make(map[string]CompareWith)
+		ctrl.refreshRequestedAppsMutex.Unlock()
+
+		oldApp := app.DeepCopy()
+		oldApp.ResourceVersion = "1"
+		oldApp.Status.ReconciledAt = &metav1.Time{Time: time.Now().Add(-1 * time.Hour)}
+
+		newApp := oldApp.DeepCopy()
+		newApp.ResourceVersion = "2"                                // Different ResourceVersion (not a resync)
+		newApp.Status.ReconciledAt = &metav1.Time{Time: time.Now()} // Status changed
+
+		// Simulate UpdateFunc call
+		simulateUpdateFunc(oldApp, newApp)
+
+		// Check that refresh was NOT requested (status-only update should be skipped)
+		checkRefreshRequested(app.QualifiedName(), false, "Status-only update without annotation")
+	})
+
+	t.Run("Status-only update WITH refresh annotation SHOULD trigger refresh", func(_ *testing.T) {
+		// Reset refresh state
+		ctrl.refreshRequestedAppsMutex.Lock()
+		ctrl.refreshRequestedApps = make(map[string]CompareWith)
+		ctrl.refreshRequestedAppsMutex.Unlock()
+
+		oldApp := app.DeepCopy()
+		oldApp.ResourceVersion = "3"
+		oldApp.Status.ReconciledAt = &metav1.Time{Time: time.Now().Add(-1 * time.Hour)}
+
+		newApp := oldApp.DeepCopy()
+		newApp.ResourceVersion = "4"                                // Different ResourceVersion
+		newApp.Status.ReconciledAt = &metav1.Time{Time: time.Now()} // Status changed
+		// Add refresh annotation
+		if newApp.Annotations == nil {
+			newApp.Annotations = make(map[string]string)
+		}
+		newApp.Annotations[v1alpha1.AnnotationKeyRefresh] = string(v1alpha1.RefreshTypeNormal)
+
+		// Simulate UpdateFunc call
+		simulateUpdateFunc(oldApp, newApp)
+
+		// Check that refresh WAS requested (annotation should trigger refresh)
+		checkRefreshRequested(app.QualifiedName(), true, "Status-only update WITH refresh annotation")
+	})
+
+	t.Run("Status-only update WITH hydrate annotation SHOULD trigger refresh", func(_ *testing.T) {
```

Comment:
For completeness, can you add a test case for scenario when both the annotations are added ? Also if the annotation is removed, is it expected to trigger a referesh ? Will it be treated as a change to the Annotation ?